NDVItimeseries.R

```diff
@@ -0,0 +1,527 @@
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Load Data -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+#admin 2
+
+NDVI1admin2 <-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/NDVIMean_ts_niger (1981-2000).csv")
+
+NDVI2admin2<-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/NDVIMean_ts_niger (2001-2022).csv")
+
+#admin 3
+
+NDVI1admin3 <-read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/l3NDVIMean_ts_niger (1981-2000).csv")
+
+NDVI2admin3 <- read.csv("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/l3NDVIMean_ts_niger (2001-2022).csv")
+
+#merge datasets
+NDVIadmin2 <- dplyr::left_join(NDVI1admin2, NDVI2admin2)
+NDVIadmin3 <- dplyr::left_join(NDVI1admin3, NDVI2admin3)
+
+# geospatial
+nigerMapAdmin2 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 2 (1)/wb_niger_admin2_shapefile/niger_admin2.shp") 
+
+
+nigerMapAdmin3 <- 
+  st_read("C:/Users/riley/OneDrive/Desktop/My classes/Economics/New folder/DSPG 2022/DSPG_Sahel/Sahel Data/Level 3 (1)/NER_adm03_feb2018.shp")
+
+
+# Don't show scientific notation
+options(scipen = 999)
+
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+# Clean titles and data -----
+# %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("_NDVI", "", colnames(NDVIadmin2)[col])
+}
+for ( col in 1:ncol(NDVIadmin2)){
+  colnames(NDVIadmin2)[col] <- sub("X", "", colnames(NDVIadmin2)[col])
+}
+
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("_NDVI", "", colnames(NDVIadmin3)[col])
+}
+for ( col in 1:ncol(NDVIadmin3)){
+  colnames(NDVIadmin3)[col] <- sub("X", "", colnames(NDVIadmin3)[col])
+}
+head(NDVIadmin3)
+
+
+# Convert data from Wide to Long ---- 
+ndviData2 <- 
+  NDVIadmin2 %>%
+  select(admin1Name, admin1Pcod, 
+         admin2Name, admin2Pcod, "19810701": "20191231") 
+
+ndviDataLong2 <- 
+  gather(ndviData2, 
+         key = Date, 
+         value = ndvi, 
+         "19810701":"20191231") %>% 
+  mutate(Date = ymd(Date)) #convert from char to dates
+# "19810701":"20191231") 
+
+ndviDataLong2$ndvi <- as.numeric(ndviDataLong2$ndvi) 
+ndviDataLong2 <- ndviDataLong2 %>% 
+  mutate(newNDVI = 0.0001 *ndvi)
+
+ndviData3 <-
+  NDVIadmin3 %>%
+  select(adm_03, rowcacode3,  
+         "19810701":"20191231")
+
+ndviDataLong3 <-
+  gather(ndviData3,
+         Date,
+         ndvi,
+         "19810701":"20191231") %>% 
+  mutate(Date = ymd(Date)) # Create new col with dates instead of chars
+
+ndviDataLong3$ndvi <- as.numeric(ndviDataLong3$ndvi) 
+ndviDataLong3 <- ndviDataLong3 %>% 
+  mutate(newNDVI = 0.0001 *ndvi)
+
+data_admin2 <- 
```

Comment:
somewhere around here it'd be useful to show some summary statistics of the values you have to see if they're in the range you're expecting prior to plotting. especially since we saw some very unusual values for the peak ndvi, would be useful to see if there's a special code for the max (like 9999 or so) that may be throwing off the calculations